nengo_dl/tests/test_simulator.py

```diff
@@ -1323,6 +1323,57 @@ def test_get_nengo_params(Simulator, seed):
         assert np.allclose(sim.data[p], sim2.data[p2])
 
 
+@pytest.mark.parametrize("use_scipy", (True, False))
+def test_get_nengo_params_sparse(Simulator, use_scipy, rng):
+    if use_scipy:
+        scipy_sparse = pytest.importorskip("scipy.sparse")
+
+    with nengo.Network() as net:
+        node0 = nengo.Node([1] * 10)
+        node1 = nengo.Node(size_in=10)
+        data = rng.uniform(-1, 1, size=10)
+        if use_scipy:
+            conn = nengo.Connection(
+                node0,
+                node1,
+                synapse=None,
+                transform=nengo.Sparse(
+                    (10, 10),
+                    init=scipy_sparse.coo_matrix(
+                        (data, (np.arange(10), np.arange(10)))
+                    ),
+                ),
+            )
+        else:
+            conn = nengo.Connection(
+                node0,
+                node1,
+                synapse=None,
+                transform=nengo.Sparse(
+                    (10, 10), indices=[[i] * 2 for i in range(10)], init=data
+                ),
+            )
+
+        p0 = nengo.Probe(node1)
+
+    with Simulator(net) as sim0:
+        params = sim0.get_nengo_params(conn)
+        assert np.allclose(params["transform"].init.data, data)
```

Comment:
Will this work if Scipy is not installed? In that case, what will `params["transform"].init` be? I'm assuming a Numpy array, but right now when I try without Scipy I'm getting `Aborted (core dumped)`. I'll look into it a bit more. Is this expected to work without Scipy, or does NengoDL only support sparse with Scipy (in which case, we should mark this whole test as requiring Scipy)?

Reply:
This test should be skipped if scipy isn't installed (in the `use_scipy=True` condition). In the `use_scipy=False` condition, `.init.data` will be `SparseMatrix.data`.

Reply:
I forgot we had the `SparseMatrix`. Anyway, I reinstalled and uninstalled Scipy and the weird error disappeared, so who knows? But it seems good now.